source/mainnet/tutorials/using-ID-in-dApps/wallet-connectors-tutorial.rst

```diff
@@ -0,0 +1,644 @@
+* **Verifier**: The application or service that needs to validate the proof (our dApp or its backend)
+
+It's important to understand that the **challenge** must be generated by the **verifier** (not the prover) to prevent replay attacks.
+Here's how the process works:
+
+1. The **verifier** generates a unique challenge (in our implementation, we use a recent block hash combined with a context string)
+2. The **prover** (wallet) creates a proof using this challenge
+3. The **verifier** checks that:
+
+  * The proof is cryptographically valid
+  * The proof was created using the specific challenge it provided
+  * The **challenge** has not been used in a previous proof
+
+Note that protection against replay attacks isn't automatic - the **verifier** must implement proper challenge validation, typically by:
+
+  * Checking that the **challenge** includes a recent block hash (time-limiting the proof)
+  * Tracking which **challenges** have been used in a database
+  * Rejecting proofs with previously-used challenges
+
+As an example, in the `compliant-reward-distribution dApp <https://github.com/Concordium/concordium-dapp-examples/blob/main/compliant-reward-distribution/indexer-and-server/src/bin/server.rs#L462>`_,
+the verifier checks that the proof hasn't expired by verifying that the block height is recent enough, but it doesn't track which challenges have been used before.
+For complete security in a production environment, a backend would need to maintain a record of used challenges to fully protect against replay attacks.
+
+We serialize these parameters using ``JSONBigInt`` instead of standard JSON. This is important because ZK proofs often involve large numbers that standard JSON can't handle correctly.
+
+Now, let's implement the actual request to the mobile wallet:
+
+.. code-block:: javascript
+
+  // Request ZK proof via WalletConnect
+  async requestVerifiablePresentation(
+    challenge: HexString,
+    statement: CredentialStatements,
+  ): Promise<VerifiablePresentation> {
+
+  // code omitted from previous code block
+
+  try {
+    // Send request to the mobile wallet
+    const result = await this.client.request<{
+      verifiablePresentationJson: string;
+    }>({
+      topic: this.topic,
+      request: {
+        method: "request_verifiable_presentation",
+        params: { paramsJson: serializedParams },
+      },
+      chainId: "ccd:testnet", // Use "ccd:testnet" for testnet or "ccd:mainnet" for mainnet
+    });
+    // Parse the result into a VerifiablePresentation
+    return VerifiablePresentation.fromString(
+      result.verifiablePresentationJson,
+    );
+  }
+
+The request is sent to the mobile wallet using the WalletConnect protocol. We specify:
+
+* ``topic``: The current session identifier
+* ``method``: The Concordium-specific ``request_verifiable_presentation`` method, which is stored in the ``METHOD_GENERATE_ZK_PROOF`` constant
```

Comment:
The constant is not visible in the code snippet anymore (so people could get confused what you mean by that constant).

```suggestion
* ``method``: The Concordium-specific ``request_verifiable_presentation`` method.
```